embedded/src/main/java/org/wildfly/core/embedded/EmbeddedServerFactory.java

```diff
@@ -293,8 +296,14 @@ private static ModuleLoader setupModuleLoader(String modulePath, String... syste
             }
             WildFlySecurityManager.setPropertyPrivileged("jboss.modules.system.pkgs", packages.toString());
 
-            // Get the module loader
-            return Module.getBootModuleLoader();
+            // setup the module class so we can reload it and reset the static properties later if we need to
+            URL[] urls = new URL[] {new File(modulePath).toURI().toURL(), new File(modulePath + "/../jboss-modules.jar").toURI().toURL()};
```

Comment:
The callers of this method should know where the root is. The EmbeddedServerFactory.create methods take separate params for jbossHomePath and modulePath, so the relationship between modulePath and jboss-modules.jar may not be fixed like this. But the relationship between jbossHomePath and jboss-modules.jar should be.